src/main/java/in/projecteka/consentmanager/dataflow/DataFlowRequestRepository.java

```diff
@@ -14,6 +14,8 @@
             "hip_id FROM consent_artefact WHERE consent_artefact_id=$1";
     private static final String INSERT_TO_HEALTH_INFO_NOTIFICATION = "INSERT INTO health_info_notification " +
             "(transaction_id, notification_request) VALUES ($1, $2)";
+    private static final String SELECT_CONSENT_ARTEFACT_EXPIRY_DATE = "SELECT consent_artefact -> 'permission' ->> 'dataEraseAt' as consent_expiry_date " +
+            "FROM consent_artefact WHERE consent_artefact_id=$1";
     private PgPool dbClient;
 
     public DataFlowRequestRepository(PgPool pgPool) {
```

Comment:
This is wrong, data flow module should not query consent service table at all. If you remove the validation in the listener and keep the validation at controller, you will not have this.

Reply:
Removed. Validation is already done at the controller level. Removed this additional check here.